### PR TITLE
Relax read-only mounts even more

### PR DESCRIPTION
**Pull request by DaanDeMeyer**
Turns out having home directories as a subdirectory of /usr is a thing. Let's relax the readonly mount requirements even more to make this use case work as well.

### CODE REVIEW

**Comment by bjackman** [comment]
Hey Daan, thanks for working to support my perverted host system :D 

So far I am still getting the same issue:

```
‣ Copying repository metadata
Traceback (most recent call last):
  File "/usr/local/google/home/jackmanb/src/mkosi/mkosi/sandbox.py", line 860, in <module>
  File "/usr/local/google/home/jackmanb/src/mkosi/mkosi/sandbox.py", line 825, in main
  File "/usr/local/google/home/jackmanb/src/mkosi/mkosi/sandbox.py", line 455, in execute
  File "<frozen os>", line 215, in makedirs
  File "<frozen os>", line 215, in makedirs
  File "<frozen os>", line 215, in makedirs
  [Previous line repeated 4 more times]
  File "<frozen os>", line 225, in makedirs
OSError: [Errno 30] Read-only file system: 'newroot/usr/local/google'
‣ "cp --recursive --no-dereference --preserve=mode,links,timestamps,ownership,xattr --reflink=auto --copy-contents '/work/usr/local/google/home/jackmanb/.cache/mkosi/debian~testing~x86-64/cache/apt' '/work/usr/local/google/home/jackmanb/src/mkosi-kernel/mkosi.cache/debian~testing~x86-64.metadata.cache/cache/apt' --no-target-directory" returned non-zero exit code 1.
```

I'm yet clear on what exactly is the issue though

**Comment by DaanDeMeyer** [comment]
> I'm not yet clear on what exactly is the issue though

mkosi is full of assumptions that /usr is read-only and we should never ever touch it. You're now encountering all of those assumptions basically.

File: mkosi/sandbox.py [hunk 1]
```diff
@@ -392,6 +392,11 @@ def is_relative_to(one: str, two: str) -> bool:
     return os.path.commonpath((one, two)) == two
 
 
+def makedirs(path: str, mode: int = 511, exist_ok: bool = True) -> None:
+    if not os.path.exists(path):
```
Comment (behrmann) [inline mkosi/sandbox.py:396]: Why the existence check, does `os.makedirs` touch the mode otherwise?

File: mkosi/sandbox.py [hunk 2]
```diff
@@ -392,6 +392,11 @@ def is_relative_to(one: str, two: str) -> bool:
     return os.path.commonpath((one, two)) == two
 
 
+def makedirs(path: str, mode: int = 511, exist_ok: bool = True) -> None:
```
Comment (behrmann) [inline mkosi/sandbox.py:395]: Slightly better readability
```suggestion
def makedirs(path: str, mode: int = 0o777, exist_ok: bool = True) -> None:
```

**Comment by DaanDeMeyer** [comment]
This should all be uncontroversial so I'll merge it already. Can address any late reviews in another PR

File: mkosi/__init__.py [hunk 3]
```diff
@@ -3283,23 +3287,29 @@ def make_extension_image(context: Context, output: Path) -> None:
...
+            cmdline += ["--private-key", context.config.verity_key]
     if context.config.verity_certificate:
-        cmdline += ["--certificate", context.config.verity_certificate]
-        options += ["--ro-bind", context.config.verity_certificate, context.config.verity_certificate]
+        cmdline += ["--certificate", workdir(context.config.verity_certificate)]
+        options += [
+            "--ro-bind",
+            context.config.verity_certificate,
+            workdir(context.config.verity_certificate),
+        ]  # noqa
```
Comment (behrmann) [inline mkosi/__init__.py:3312]: This noqa doesn't seem to be necessary? At least, removing it doesn't give me any warnings.

File: mkosi/__init__.py [hunk 4]
```diff
@@ -2999,7 +3003,7 @@ def make_image(
             "--ro-bind",
             context.config.verity_certificate,
             workdir(context.config.verity_certificate),
-        ]
+        ]  # noqa
```
Comment (behrmann) [inline mkosi/__init__.py:3006]: Removing this one also doesn't trigger any warnings for me.

**Pull request merged**